### ISSUE TITLE
generic.secrets.security.detected-npm-registry-auth-token.detected-npm-registry-auth-token triggers for auth token in env variable

### ISSUE THREAD

**Issue by Akshay090** [returntocorp/sgrep-rules#2530]
**Describe the bug**
A clear and concise description of what the bug is and for which rule. Is the rule producing a false positive? Is it incorrect or missing something?

👉🏻 `generic.secrets.security.detected-npm-registry-auth-token.detected-npm-registry-auth-token
` falsely triggers when Auth Token is passed as env value 

**To Reproduce**
Sample code to reproduce this behavior.

.npmrc
```
//registry.npmjs.org/:_authToken=${NPM_TOKEN}
```

 Passing NPM TOKEN as env value should not trigger this rule, this approach is also suggested in official npm docs - https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow

**Expected behavior**

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**

**Comment by 0xkrishna17** [comment]
✔️
